### PR TITLE
commons-parent: small cleanups to pom.xml

### PR DESCRIPTION
**Pull request by grimreaper**
This is multiple commits and can be separately merged.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -118,58 +118,51 @@
       <subscribe>user-subscribe@commons.apache.org</subscribe>
       <unsubscribe>user-unsubscribe@commons.apache.org</unsubscribe>
       <post>user@commons.apache.org</post>
-      <archive>http://mail-archives.apache.org/mod_mbox/commons-user/</archive>
+      <archive>https://mail-archives.apache.org/mod_mbox/commons-user/</archive>
```
Comment (lukaszlenart) [inline pom.xml]: Wouldn't be better to use https://lists.apache.org/list.html?user@commons.apache.org instead? It's a more user-friendly solution.

File: pom.xml [hunk 2]
```diff
@@ -118,58 +118,51 @@
...
+        <otherArchive>https://markmail.org/list/org.apache.commons.commits/</otherArchive>
+        <otherArchive>https://www.mail-archive.com/commits@commons.apache.org/</otherArchive>
       </otherArchives>
     </mailingList>
     <mailingList>
       <name>Apache Announce List</name>
       <subscribe>announce-subscribe@apache.org</subscribe>
       <unsubscribe>announce-unsubscribe@apache.org</unsubscribe>
-      <archive>http://mail-archives.apache.org/mod_mbox/www-announce/</archive>
+      <archive>https://mail-archives.apache.org/mod_mbox/www-announce/</archive>
```
Comment (lukaszlenart) [inline pom.xml]: Same here https://lists.apache.org/list.html?dev@commons.apache.org ?

File: pom.xml (hunk 1 above)
Comment (grimreaper) [reply pom.xml]: I was unaware of that option. Will note for the future.

File: pom.xml (hunk 2 above)
Comment (grimreaper) [reply pom.xml]: `dev@` isn't the same as `announce@` but I get the point.

Comment (lukaszlenart) [reply pom.xml]: Sorry, I meant `dev@`, but I put this comment in wrong place.

**Pull request merged**